js-code-guidelines.md

```diff
@@ -73,39 +69,78 @@ For the majority of our JavaScript projects, our goals are to:
+- [David Dias](http://github.com/diasdavid/) for js-ipfs, js-libp2p js-multiformats ecosystems.
+- [Volker Mische](github.com/vmx) for the js-ipld ecosystem.
 
-Sometimes, a Captain may elect to have other maintainers that also have merge ability and commit access to the main repo. These maintainers can help out, but defer to the Captain as the person in charge of maintaining quality across the repo. It is important that this distinction is explicit; if there are long-standing PRs or issues, it is ultimately up to the Captain to gather information about the issue or PR. A Captain only makes a decision if he _needs_ to and _all_ methods of discussion are exhausted. Our community strives to trust the Captain as someone who ultimately has the most knowledge of a repo (even if they are also opinionated, and even if they have to spend effort to source that knowledge). This may change in the future, if we go with more non-hierarchical model.
+The current Lead Maintainers can be identified either by the `maintainer` field in the package.json of the module and/or the section `Maintainer` in the README.md of the module.
 
-Repositories which do not have Captains will show up in red on [Project Repos](https://project-repos.ipfs.io). Please open an issue in the repository asking for a captain to be nominated if none is provided.
+**Lead Maintainer responsibilities:**
 
-If you would like to become a maintainer (or lieutenant, or first mate, or whatever nautical metaphor you prefer), please open an issue about it! Chances are that we would love to have more help.
+- Respect and follow the [IPFS Code of Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
+- Have a complete understanding of the module purpose, its specification (if any) and how the module is used by other parts of the project.
+- Review and Release PRs
+- Publish new versions of the module to npm
 
-## Guidelines
+**Lead Maintainer expected attitude:**
 
-### Linting & Code Style
+- Be proactive in increasing the quality of the module. This goes from improving documentation, tests, codebase and more.
+- Show a great level of rigor and polish in the code that they ship.
+- Help others in understanding how the module and why it exists.
+- Apply the established Contributing Guidelines to the project and help others do too
+
+**How to become a Lead Maintainer:**
+
+Currently, there is no formal test or request form to do so. Lead Maintainers will be invited and nominated by the Tech Lead once the expected behaviour and rigor is observed. A Lead Maintainer can nominate the next Lead Maintainer to the Tech Lead 
+
+**In practice:**
+
+- Update each package.json and README.md to have a `maintainer` field.
+- Update packages table on each entry module (e.g https://github.com/ipfs/js-ipfs#packages) to also list the maintainer for each.
+- Protect the master branch and only grant permissions for merge to the Maintainer and the Tech Lead, only.
```

Comment:
This sounds like merge and publish permissions will be limited to 2 people. Has there been any thought about expanding that for larger modules? I see the value around module integrity and quality with a smaller set of responsible individuals, but there is also the risk of only 2 people becoming blockers if they are unavailable during the same period of time (conferences, retreats, vacations, etc).

Reply:
Fair point, that said, do note that this is 100% for almost every module. I expect that there there will be a few or almost no situation where there is a need for a urgent release without the Tech Lead and the Lead Maintainer being around.

In the case that one or both of the Tech Lead and Lead Maintainer need to be absent for a long period of time, both of these should notify the rest of the group and a plan for releasing during that time should be prepared. My take is that if someone takes a week or two offline during a quiet period, it is almost certainly ok to wait 2 weeks to release, the only exception would be a security fix but those need to be handled with care and in with the the Security team.